plat/nvidia/tegra/common/tegra_bl31_setup.c

```diff
@@ -226,3 +227,35 @@ void bl31_plat_arch_setup(void)
 	/* enable the MMU */
 	enable_mmu_el3(0);
 }
+
+/*******************************************************************************
+ * Check if the given NS DRAM range is valid
+ ******************************************************************************/
+int bl31_check_ns_address(uint64_t base, uint64_t size_in_bytes)
+{
+	uint64_t end = base + size_in_bytes;
```

Comment:
If size_in_bytes is close to the maximum UINT64 value then this calculation will wrap with the result of end < base. This will break various of the bounds checks. The "base == end" check in line 242 should check for (base >= end) to also detect this invalid parameter.